src/js/editor/post.js

```diff
@@ -720,6 +721,30 @@ class PostEditor {
     return this.moveSectionBefore(collection, renderedSection, beforeSection);
   }
 
+  // TODO write tests for this
+  insertMarkers(position, markers) {
+    let { section, offset } = position;
+    assert('Cannot insert markers at non-markerable position',
+           section.isMarkerable);
+
+    let edit = section.splitMarkerAtOffset2(offset);
+    edit.removed.forEach(marker => this._scheduleForRemoval(marker));
+    edit.added.forEach(marker => this._markDirty(marker));
```

Comment:
Added things should not need to be marked dirty, they are already dirty. `this._markDirty(section)` as you have below is sufficient.